metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java

```diff
@@ -178,4 +188,44 @@ private boolean isShortDoi(String value) {
         return value.matches(SHORT_DOI_REGEX);
     }
 
+    public String lookUpJournalIdAtPublicationChannel(String name, String electronicIssn, String printedIssn, int year)
+            throws IOException, InterruptedException {
+        String id;
+        if (StringUtils.isNotEmpty(electronicIssn)) {
+            id = queryPublicationChannelForJournal(electronicIssn, year);
+            if (StringUtils.isNotEmpty(id)) {
+                return id;
+            }
+        }
+        if (StringUtils.isNotEmpty(printedIssn)) {
+            id = queryPublicationChannelForJournal(printedIssn, year);
+            if (StringUtils.isNotEmpty(id)) {
+                return id;
+            }
+        }
+        if (StringUtils.isNotEmpty(name)) {
+            id = queryPublicationChannelForJournal(name, year);
+            if (StringUtils.isNotEmpty(id)) {
+                return id;
+            }
+        }
+        return null;
+    }
+
+    private String queryPublicationChannelForJournal(String term, int year) throws IOException, InterruptedException {
+        String query =  "query=" + URLEncoder.encode(term, StandardCharsets.UTF_8.toString()) + "&year=" + year;
+        String uri = API_URI_NVA_PUBLICATION_CHANNELS + "/journal?" + query;
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .headers("Accept", "application/ld+json")
+                .GET()
+                .build();
+        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        if (SC_OK == response.statusCode()) {
+            JSONArray jsonArray = new JSONArray(response.body());
+            JSONObject jsonObject = jsonArray.getJSONObject(0);
```

Comment:
Here we say explicitly that we want the first element from the array, make this clear extracting a method with a suitable name "getIdOfFirstElement"

Reply:
done